src/models/student-groups/actions.js

```diff
@@ -0,0 +1,97 @@
+import {
+  call, put, takeLatest, all, fork, takeEvery,
+} from 'redux-saga/effects';
+import { ApiService } from '../../shared/api-service/index.js';
+import * as actionTypes from './action-types.js';
+
+export const globalLoadStudentGroups = () => ({
+  type: actionTypes.LOAD_STUDENT_GROUPS,
+});
+
+function* loadStudentGroupsAsync() {
+  try {
+    yield put({ type: actionTypes.LOADING_STUDENT_GROUPS_STARTED });
+
+    const groups = yield call(ApiService.load, '/student_groups');
+
+    yield put({ type: actionTypes.LOADING_STUDENT_GROUPS_SUCCEED, payload: groups });
+  } catch (e) {
+    yield put({ type: actionTypes.LOADING_STUDENT_GROUPS_FAILED });
+  }
+}
+
+function* watchLoadingStudentGroups() {
+  yield takeLatest(actionTypes.LOAD_STUDENT_GROUPS, loadStudentGroupsAsync);
+}
+
+export const loadStudentGroupsById = (id) => ({
+  type: actionTypes.LOAD_STUDENT_GROUP_BY_ID,
+  payload: id,
+});
+
+function* loadStudentGroupsByIdAsync({ payload }) {
```

Comment:
No need to create additional method to get a specific group by id. You can take this group from all groups

Reply:
Sure, but the model is a representation of the backend API. You can use both the saga effect and pre-loaded store.